src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -122,8 +125,8 @@ public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionK
             }
 
             _http2Enabled = _poolManager.Settings._maxHttpVersion >= HttpVersion.Version20;
-            // TODO: Replace with Platform-Guard Assertion Annotations once https://github.com/dotnet/runtime/issues/44922 is finished
-            if ((OperatingSystem.IsLinux() && !OperatingSystem.IsAndroid()) || OperatingSystem.IsWindows() || OperatingSystem.IsMacOS())
+
+            if (_http3Enabled)
```

Comment:
This field doesn't really look like set with OS platform check result, not sure if this is the intended usage, probably i just revert all changes in `HttpConnectionPool.cs` instead and let you guys handle it. CC @marek-safar @ManickaP

Reply:
Change to if (_http3Enabled) does not make much sense. I think @ManickaP wanted to write something like 

```c#
if (IsHttpSupported) {
// existing code
}

[SupportedOSPlatformGuard("linux")]
[UnSupportedOSPlatformGuard("android")]
[SupportedOSPlatformGuard("windows")]
[SupportedOSPlatformGuard("osx")]
private static bool IsHttpSupported => (OperatingSystem.IsLinux() && !OperatingSystem.IsAndroid()) || OperatingSystem.IsWindows() || OperatingSystem.IsMacOS();

```

Reply:
`_http3Enabled` is not suitable to be used here, it's another layer (even if HTTP/3 is technically supported it still might not be enabled - AppSwitch for disabling / non-SSL connection / proxy connection etc.)

What @marek-safar suggests is more like what we need.

Also there's 6 occurrences in the code base, you're replacing only 3 of them.

Reply:
And we also need to check if the linker still trims the code after the change.

Reply:
> What @marek-safar suggests is more like what we need.

I thought its need, but I did not add that because it seems to need some other check as you mentioned, but those are can be set within this method so i will just add that checks to the field.

> Also there's 6 occurrences in the code base, you're replacing only 3 of them.

Yes, I saw them, they are in a different class and have no separate field or member used as a guard, just using the platform checks once, so doesn't seem to need updating with existing code. Overall It seems there is a public API for http3 support, probably `QuicImplementationProvider.IsSupported` and can be used as a guard instead. https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/QuicImplementationProvider.cs#L12 
But seems its implementation not completed yet, that is why i meant to leave the guard annotation for System.Net.Http. out of this PR, anyway, for now, I can create a private field (with the same platform check) for other classes and use them as a guard.